subprojects/model-core/src/main/java/org/gradle/internal/reflect/annotations/impl/DefaultTypeAnnotationMetadataStore.java

```diff
@@ -670,4 +645,35 @@ public String getDisplayName() {
             return displayName;
         }
     }
+
+    private class IrrelevantMethods implements Predicate<Method> {
+
+        @Override
+        public boolean test(Method method) {
+            if (method.isSynthetic()) {
+                return true;
+            }
+            if (method.isBridge()) {
+                return true;
+            }
+            if (isIgnoredGeneratedGroovyMethod(method)) {
+                return true;
+            }
+            return false;
+        }
+
+        /**
+         * Groovy getters are annotated with `@Generated` but shouldn't be ignored.
+         * Other methods which have '$' in their name are usually a good heuristic for the
+         * fact we should ignore them, for example methods generated by the @Memoized annotation
+         * ref: https://github.com/gradle/gradle/issues/16117
+         *
+         * @param method a candidate method
+         * @return true if we suspect this method to be a safely ignorable generated method
+         */
+        private boolean isIgnoredGeneratedGroovyMethod(Method method) {
+            return generatedMethodDetector.test(method) && method.getName().contains("$");
```

Comment:
Given that you use the `generatedMethodDetector`, I suppose you don't need to move this to a separate class, right?

Reply:
I don't _have to_, no.

Reply:
I think the code is easier to read if you don't, so I suggest you only use `generatedMethodDetector` and keep the current methods in place.